File: cpp/mouse.cpp
#include <iostream>
#include <chrono>
#include <thread>

#ifdef _WIN32
#include <Windows.h>
#else
#include <X11/Xlib.h>
#include <X11/extensions/XTest.h>
#endif

class MouseController {
private:
    static const int START_HOUR = 8;
    static const int END_HOUR = 17;
    static const int ONE_MINUTE = 60000;
    static const int TEN_SECONDS = 10000;

public:
    void moveMouse(int x, int y) {
#ifdef _WIN32
        SetCursorPos(x, y);
#else
        Display* display = XOpenDisplay(nullptr);
        XWarpPointer(display, None, DefaultRootWindow(display), 0, 0, 0, 0, x, y);
        XFlush(display);
        XCloseDisplay(display);
#endif
    }

    void run() {
        auto startTime = std::chrono::system_clock::now();

        while (true) {
            auto currentTime = std::chrono::system_clock::now();
            auto elapsedTime = std::chrono::duration_cast<std::chrono::milliseconds>(currentTime - startTime).count();

            if (elapsedTime >= ONE_MINUTE) {
                checkMouseMovement();
                startTime = currentTime;
            }

            std::this_thread::sleep_for(std::chrono::milliseconds(TEN_SECONDS));
        }
    }

private:
    void checkMouseMovement() {
        // Implementation for checking mouse movement goes here
    }
};

int main() {
    MouseController mouseController;
    mouseController.run();

    return 0;
}

File: node/mouse.js
const robot = require('robotjs');

const START_HOUR = 8;
const END_HOUR = 17;
const ONE_MINUTE = 60000;
const TEN_SECONDS = 10000;

function moveMouse(x, y) {
    robot.moveMouseSmooth(x, y);
}

function run() {
    const startTime = new Date();

    setInterval(() => {
        const currentTime = new Date();
        const elapsedTime = currentTime - startTime;

        if (elapsedTime >= ONE_MINUTE) {
            // One minute has passed, perform mouse movement check
            console.log("Checking mouse movement...");
            startTime = currentTime;
        }
    }, TEN_SECONDS);
}

// Example usage of moveMouse function
moveMouse(100, 100);

run();
